specification/search/data-plane/Azure.Search/preview/2019-05-06-preview/searchservice.json

```diff
@@ -4296,6 +4296,51 @@
         "url": "https://lucene.apache.org/core/4_10_3/analyzers-common/org/apache/lucene/analysis/pattern/PatternReplaceCharFilter.html"
       }
     },
+    "Similarity": {
+      "discriminator": "@odata.type",
+      "properties": {
+        "@odata.type": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "@odata.type"
+      ],
+      "description": "Base type for similarity algorithms. Similarity algorithms are used to calculate scores that tie queries to documents. The higher the score, the more relevant the document is to that specific query. Those scores are used to rank the search results.",
+      "externalDocs": {
+        "url": "https://docs.microsoft.com/azure/search/index-ranking-similarity"
+      }
+    },
+    "ClassicSimilarity": {
+      "x-ms-discriminator-value": "#Microsoft.Azure.Search.ClassicSimilarity",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Similarity"
+        }
+      ],
+      "description": "Legacy similarity algorithm which uses the Lucene TFIDFSimilarity implementation of TF-IDF. This variation of TF-IDF introduces static document length normalization as well as coordinating factors that penalize documents that only partially match the searched queries."
+    },
+    "BM25Similarity": {
+      "x-ms-discriminator-value": "#Microsoft.Azure.Search.BM25Similarity",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Similarity"
+        }
+      ],
+      "properties": {
+        "k1": {
+          "type": "number",
+          "format": "double",
+          "description": "The k1 parameter controls the scaling function between the term frequency of each matching terms and the final relevance score of a document-query pair. By default, a value of 1.2 is used. A value of 0.0 means the score does not scale with an increase in term frequency."
```

Comment:
I would not reference the name of the parameter/property since it will end up in generated code and we'd have to override in code just to change the description. Maybe instead something like, "This property controls ...".

Reply:
Same for instances below.

Reply:
Thanks. What about the description for the BM25Similarity model? I reference both k1 and b by name (since they are part of the official definition of the BM25 formula).

Reply:
I don't see a way around that. Since they're part of the official definition of BM25, maybe it's okay to leave them in that instance.